### DISCUSSION TITLE
Kafka UI url gives connection refused error

### DISCUSSION THREAD

**Discussion by bharathiraja1986** [provectus/kafka-ui#4070] [Q&A] [answered]
Hi Team,
We have been working to enable the kafka ui for our AWS kafka cluster.  Currently, we are facing issue with SSL configuration where it is giving 'connection refused error' on hitting the UI url. Backend, we could see the process started without any error and able to get the metrics from Kafka clusters. But facing problem only in accessing the UI. 
We followed the below SSL configuration page. Could you please help us on this ?
https://github.com/provectus/kafka-ui/blob/master/documentation/compose/kafka-ssl.yml

current configuration for UI:

version: '3'
services:

  kafka-ui:
    container_name: kafka-ui
    image: provectuslabs/kafka-ui
    ports:
      - 443:8080
    environment:
      SERVER_PORT: 8333
      KAFKA_CLUSTERS_0_NAME: local
      KAFKA_CLUSTERS_0_PROPERTIES_SECURITY_PROTOCOL: SSL
      KAFKA_CLUSTERS_0_PROPERTIES_SSL_KEYSTORE_TYPE: JKS
      KAFKA_CLUSTERS_0_PROPERTIES_SSL_KEYSTORE_LOCATION: /kafka.keystore.jks
      KAFKA_CLUSTERS_0_PROPERTIES_SSL_KEYSTORE_PASSWORD: "sometext"
      KAFKA_CLUSTERS_0_PROPERTIES_SSL_TRUSTSTORE_TYPE: JKS
      KAFKA_CLUSTERS_0_SSL_TRUSTSTORELOCATION: /kafka.truststore.jks
      KAFKA_CLUSTERS_0_SSL_TRUSTSTOREPASSWORD: "sometext"
      KAFKA_CLUSTERS_0_PROPERTIES_SSL_ENDPOINT_IDENTIFICATION_ALGORITHM: ''
      DYNAMIC_CONFIG_ENABLED: true
    volumes:
      - /apps/config/dynamic_config.yaml:/etc/kafkaui/dynamic_config.yaml
      - /apps/ssl/kafka.keystore.jks:/kafka.keystore.jks
      - /apps/ssl/kafka.truststore.jks:/kafka.truststore.jks

**Comment by Haarolean** [comment]
Hi, can't help you without proper details. "SSL protocol Error" is a generic error message, doesn't say anything specific.

**Reply by bharathiraja1986** [reply]
it is giving connection refused error now but application is up and running in the container . command used,

curl -vk https://*****:8333/actuator

ALPN, offering h2
* ALPN, offering http/1.1
* successfully set certificate verify locations:
*   CAfile: /etc/pki/tls/certs/ca-bundle.crt
  CApath: none
* TLSv1.3 (OUT), TLS handshake, Client hello (1):
* error:1408F10B:SSL routines:ssl3_get_record:wrong version number
* Closing connection 0
curl: (35) error:1408F10B:SSL routines:ssl3_get_record:wrong version number

**Comment by Haarolean** [comment]
[kafka-ssl.yml](https://github.com/provectus/kafka-ui/blob/master/documentation/compose/kafka-ssl.yml) is about setting up SSL for kafka, not kafka-ui.
For SSL on kafka-ui, use either a reverse proxy (recommended) or [this (3.1)](https://www.baeldung.com/spring-boot-https-self-signed-certificate)

**Reply by bharathiraja1986** [reply, marked as answer]
We fixed it by changing docker expose port and adding ip address to it. Also there was one more jwt token url issue . We updated it to reflect value from okta JWT
